src/gaxios.ts

```diff
@@ -87,17 +57,50 @@ export class Gaxios {
    * @param opts Set of HTTP options that will be used for this HTTP request.
    */
   async request<T = any>(opts: GaxiosOptions = {}): GaxiosPromise<T> {
-    opts = await this.#prepareRequest(opts);
-    return this._request(opts);
+    const prepared = await this.#prepareRequest(opts);
+    return this._request(prepared);
   }
 
   private async _defaultAdapter<T>(
-    opts: GaxiosOptions
+    config: GaxiosOptions
   ): Promise<GaxiosResponse<T>> {
-    const fetchImpl = opts.fetchImplementation || fetch;
-    const res = (await fetchImpl(opts.url, opts)) as FetchResponse;
-    const data = await this.getResponseData(opts, res);
-    return this.translateResponse<T>(opts, res, data);
+    const fetchImpl =
+      config.fetchImplementation ||
+      this.defaults.fetchImplementation ||
+      (await Gaxios.#getFetch());
+
+    // node-fetch v3 warns when `data` is present
+    // https://github.com/node-fetch/node-fetch/issues/1000
+    const preparedOpts = {...config};
+    delete preparedOpts.data;
+
+    const res = (await fetchImpl(config.url!, preparedOpts as {})) as Response;
+    let data = await this.getResponseData(config, res);
+
+    // `node-fetch`'s data isn't writable. Native `fetch`'s is.
+    if (Object.getOwnPropertyDescriptor(res, 'data')?.configurable) {
+      // Keep `Response` as a class
+      return Object.assign(res, {config, data});
+    } else {
+      Object.assign(res, {config});
+
+      // best effort for `node-fetch`; `.data` is not writable...
+      return new Proxy(res, {
```

Comment:
There's gotta be a cleaner way to do this... hmm. We want to expose all of `res`'s types so that customers can take advantage of new features as they're released